docs/docs/running-hello-world.md

```diff
@@ -73,265 +85,289 @@ The web server loads the enclave and waits for requests from the client. Amongst
+    !!! note
+        * You need Docker to run the *host* on Windows. You don't need Docker to run the client.
+        * It is not possible to run enclaves in debug or release mode on Windows.
 
-!!! note
-    If your platform or container doesn't support SGX enclaves, you should see the following message when you run the host:
+
+    1. Create the `conclave-build` Docker image, which you can use to create a Linux environment to run the host.
+       ```bash
+       gradlew.bat enclave:setupLinuxExecEnvironment
+       ```
+    2. Instantiate a container using the `conclave-build` image using
+      ```bash
+      docker run -it --rm -p 8080:8080 -v ${PWD}:/project -w /project conclave-build /bin/bash
+      ```
+      This will give you a bash shell in the container that simulates a native Linux machine. You can tweak this command
+      according to the specific needs of your project. Please take a look at the explanation of the options used in this
+      command [at the end of this tutorial](#appendix-summary-of-docker-command-options).
+      Check the [Docker reference manual](https://docs.docker.com/engine/reference/commandline/run/) or run 
+      `docker <command> --help` on the Docker command line interface for more information.
+    3. Run the host and client as you did for mock mode. The only differences are that the host must be run from inside
+      the container, and the host JAR is named `host-simulation.jar` rather than `host-mock.jar`.
+    4. Press `CTRL+D` to exit the container when finished.
+
+    Alternatively, you can also install the Windows subsystem for Linux (2) and download Ubuntu 20.04 from the Windows
+    store. From there, you can launch an Ubuntu 20.04 shell from the start menu and build or run Conclave applications
+    as in a native Linux environment.
+    Instructions for installing WSL-2 can be found [here](https://docs.microsoft.com/en-us/windows/wsl/install).
+    Please note that this method hasn't been extensively tested.
+
+
+!!!Note
+
+    If your platform or container doesn't support SGX enclaves, you might see the following message when you run the
+    host:
     ```text
-    This platform does not support hardware enclaves: SGX_DISABLED_UNSUPPORTED_CPU: SGX is not supported by the CPU in this system
+    This platform does not support hardware enclaves: SGX_DISABLED_UNSUPPORTED_CPU: SGX is not supported by the 
+    CPU in this system
     ```
-    Don't worry, you will still be able to use simulation mode even if you see this message.
+Don't worry, you will still be able to use simulation mode even if you see this message.
 
 ### Run the client in other modes
 The client build is independent of the mode. The only difference is that the enclave will have a different signing key
-when you are not using mock mode, which will be reflected in the Enclave Constraint provided to the client.
+when you are not using mock mode, which will be reflected in the enclave constraint provided to the client.
```

Comment:
It's not just mock mde necessarily. For example, we talk about release mode have an external code signing key. That will mean it will be different from debug and simulation. So I would rephrase his sentence to something like

"The only difference is that the enclave might have a different code coder, ..."